src/agent/config.ts

```diff
@@ -17,12 +17,12 @@
 import {AuthenticationConfig} from '../types/common-types';
 
 export interface DebugAgentConfig extends AuthenticationConfig {
-  workingDirectory: string|null;
+  workingDirectory?: string;
 
   /**
    * A user specified way of identifying the service
    */
-  description: string|null;
+  description?: string;
 
   /**
    * Whether or not it is permitted to evaluate expressions.
```

Comment:
I think this would be best addressed in a follow-up PR where the config provided to the `start` method has all of its members as optional.

The default config, however, is of a different `ResolvedDebugAgentConfig` type where `allowExpressions`, `serviceContext`, and `workingDirectory` are not optional since they are resolved to specific values.

The config used throughout the code, then, is of the `ResolvedDebugAgentConfig` type where `allowExpressions`, `serviceContext`, and `workingDirectory` are not optional since their values are filled in from the default config if not specified in the `start` method.

This would simplify a lot of the type checks in the code, but I think should be in its own PR.